src/agents/base_agent.py

```
@@ -136,19 +139,37 @@ def log_data(self, train_data: pd.DataFrame, test_data: pd.DataFrame):
         self.model_kwargs["env"] = env
         model = self.model(**self.model_kwargs)
 
-        # RUN TRAINING WITH MLFLOW CALLBACK, LOG ALL TEST IN MLFLOW CALLBACK
-        model.learn(total_timesteps=self.train_timesteps, callback=model_callback)
-        # mlflow.pytorch.log_model(model, "model")
+        # TRAIN AND SAVE MODEL PERIODICALLY
+        model_name = get_model_name(self.experiment_name)
+        current_run = mlflow.active_run()
+        run_name = current_run.info.run_name
+
+        for step in range(0, self.train_timesteps, self.save_freq):
```

Ruff

Loop control variable `step` not used within loop body

Rename unused `step` to `_step`

(B007)